File: csharp/bob/Bob.cs
using System;
using System.Linq;
using System.Text.RegularExpressions;

public static class Bob
{
    public static string Response(string statement)
    {
        statement = statement.Trim();

        if (statement.Length == 0)
        {
            return "Fine. Be that way!";
        }
        else if (statement.Any(c => char.IsLetter(c))
          && statement.Where(c => char.IsLetter(c)).All(c => char.IsUpper(c)))
        {
            return "Whoa, chill out!";
        }
        else if (statement.EndsWith('?'))
        {
            return "Sure.";
        }

        return "Whatever.";
    }
}

File: php/bob/bob.php
<?php

class Bob {
    function respondTo($msg) {
        $msg = trim($msg);

        if (preg_match("/^$/", $msg)) {
            return "Fine. Be that way!";
        } elseif (preg_match("/[A-Z]/", $msg) && !preg_match("/[a-z]/", $msg)) {
            return "Whoa, chill out!";
        } elseif (preg_match("/\?$/", $msg)) {
            return "Sure.";
        } else {
            return "Whatever.";
        }
    }
}
